scripts/pylib/twister/twisterlib/platform.py

```diff
@@ -2,16 +2,44 @@
 # vim: set syntax=python ts=4 :
 #
 # Copyright (c) 2018-2022 Intel Corporation
+# Copyright (c) 2024 Arm Limited (or its affiliates). All rights reserved.
+#
 # SPDX-License-Identifier: Apache-2.0
 
 import os
+import shutil
+from typing import Dict, List, Optional
 import scl
 from twisterlib.environment import ZEPHYR_BASE
+from twisterlib.constants import SUPPORTED_SIMS
 import logging
 
 logger = logging.getLogger('twister')
 logger.setLevel(logging.DEBUG)
 
+
+class Simulator:
+    """Class representing a simulator"""
+
+    def __init__(self, data: Dict[str, str]):
+        assert "name" in data
+        assert data["name"] in SUPPORTED_SIMS
+        self.name = data["name"]
+        self.exec = data.get("exec")
+
+    def is_runnable(self) -> bool:
+        return (self.name == 'qemu') or bool(self.exec) and bool(shutil.which(self.exec))
```

Comment:
that does not work for native and other simulators, native_sim now is build_only and will never run, that is bad ;-(

Reply:
Indeed. Thank you for looking into this. I think I found the culprit and will be pushing an update once I'm done running extra checks.